crates/sui-types/src/temporary_store.rs

```diff
@@ -364,6 +377,10 @@ impl<'backing> TemporaryStore<'backing> {
             }
         });
 
+        self.formerly_deleted_shared_objects
+            .iter_mut()
+            .for_each(|obj| modified_at_versions.push(*obj));
```

Comment:
This won't work with state accumulator.
Curious, why do we want to include formerly_deleted_shared_objects in modified_at_versions?